src/access/AccessControl.cpp

```diff
@@ -352,6 +356,19 @@ CHIP_ERROR AccessControl::Check(const SubjectDescriptor & subjectDescriptor, con
         VerifyOrReturnError(requestPath.requestType != RequestType::kRequestTypeUnknown, CHIP_ERROR_INVALID_ARGUMENT);
     }
 
+#if CHIP_CONFIG_USE_ACCESS_RESTRICTIONS
+    if (mAccessRestriction != nullptr)
+    {
+        CHIP_ERROR result = mAccessRestriction->Check(subjectDescriptor, requestPath);
+        if (result != CHIP_NO_ERROR)
```

Comment:
This is dangerous - if the access restriction delegate fails for any reason, access control fails out and denies access. This needs to be a specific check because this is a swappable component that may return ANY error code.

Reply:
What would you suggest?  This is part of the overall access control check which also blocks access upon any failure (other than CHIP_ERROR_NOT_IMPLEMENTED).  Since ARL is an extension of this processing, I don't see why it should behave differently.  If ARL is activated at compile time and populated with data at runtime, I believe it should block access upon any failure (except CHIP_ERROR_NOT_IMPLEMENTED which is indeed a gap right now).

Reply:
Because this is a component that is installed via SetAccessRestriction at runtime, you don't know what's on the other side of this. It could be failing with any error, you don't really know. If the delegate fails in a generic way, all clients are shut down from access to anything on the device including access over PASE.

Reply:
Denying access is better than allowing access to things that are not allowed. Definitely though the code must be well tested to avoid this fallback to always being in error.

Reply:
I agree that going conservative is better in general, but I'm worried that if there is a generic error it shuts down access to everything on the device, including clusters that are disallowed from being restricted by the ARL.

Reply:
That is the intent of this addition though.  It is a hard check and must pass if the feature is enabled.